presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -84,9 +96,16 @@ int getOutstandingSplitCount()
     CompletableFuture<?> addToQueue(InternalHiveSplit split)
     {
         if (throwable.get() == null) {
+            if (estimatedSplitSizeInBytes.addAndGet(split.getEstimatedSizeInBytes()) > maxOutstandingSplitsBytes) {
```

Comment:
This can overflow if the limit is close to MAX_INT, and/or if there are many threads incrementing at once

Reply:
We should probably make this a long, just to be safe